File: split_blog_posts.py

import json
import os
import shutil

# Read the original file
with open('data/blog-posts.json', 'r', encoding='utf-8') as f:
    blog_posts = json.load(f)

# Create directory
output_dir = 'data/blog-posts'
os.makedirs(output_dir, exist_ok=True)

print(f"Total blog posts to split: {len(blog_posts)}")

# Create index file (without content)
index = []
for post in blog_posts:
    index.append({
        'id': post['id'],
        'title': post['title'],
        'slug': post['slug'],
        'date': post['date'],
        'category': post['category'],
        'description': post['description']
    })

with open('data/blog-index.json', 'w', encoding='utf-8') as f:
    json.dump(index, f, ensure_ascii=False, indent=2)

print("✅ Blog index file created")

# Create individual files for each post
for post in blog_posts:
    file_path = os.path.join(output_dir, f"{post['id']}.json")
    with open(file_path, 'w', encoding='utf-8') as f:
        json.dump(post, f, ensure_ascii=False, indent=2)
    print(f"✅ Created post {post['id']}: {post['title']}")

# Rename original file as backup
original_path = 'data/blog-posts.json'
backup_path = 'data/blog-posts-backup.json'
shutil.move(original_path, backup_path)
print(f"✅ Original file backed up as blog-posts-backup.json")

print("\n🎉 All tasks completed successfully!")

File: split-blog-posts.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// 读取现有数据
const blogPostsPath = path.join(__dirname, 'data', 'blog-posts.json');
const blogPosts = JSON.parse(fs.readFileSync(blogPostsPath, 'utf8'));

// 创建目录
const blogPostsDir = path.join(__dirname, 'data', 'blog-posts');
if (!fs.existsSync(blogPostsDir)) {
  fs.mkdirSync(blogPostsDir, { recursive: true });
}

// 创建索引文件（只包含元数据）
const blogIndex = blogPosts.map(post => ({
  id: post.id,
  title: post.title,
  slug: post.slug,
  date: post.date,
  category: post.category,
  description: post.description,
  featured: post.featured || false,
  // 只保留第一张图片作为预览
  thumbnail: post.images && post.images.length > 0 ? post.images[0] : undefined
}));

// 写入索引文件
fs.writeFileSync(
  path.join(__dirname, 'data', 'blog-index.json'),
  JSON.stringify(blogIndex, null, 2),
  'utf8'
);

// 写入每个独立文章文件
blogPosts.forEach(post => {
  fs.writeFileSync(
    path.join(blogPostsDir, `${post.id}.json`),
    JSON.stringify(post, null, 2),
    'utf8'
  );
});

console.log(`✅ 成功拆分 ${blogPosts.length} 篇博客文章！`);
console.log(`✅ 创建索引文件: data/blog-index.json`);
console.log(`✅ 创建文章目录: data/blog-posts/`);
console.log(`✅ 文章文件: data/blog-posts/1.json - data/blog-posts/${blogPosts.length}.json`);
